### PR TITLE
chore: add Storybook workflow doc

### PR DESCRIPTION
**Pull request by siddharthkul**
**What changed? Why?**
Added docs for storybook workflow

Screenshot 2024-07-15 at 9 02 42 PM

**How has it been tested?**
`cd site && yarn dev`

### CODE REVIEW

File: site/docs/pages/guides/contribution.mdx [hunk 1]
```diff
@@ -65,6 +65,18 @@ onchainkit
...
+### Storybook
+
+Storybook is a frontend workshop for building UI components and pages in isolation. It helps you develop and share hard-to-reach states and edge cases without needing to run your whole app.
+
+To develop and test your components in Storybook, run the following command:
+
+```bash
+yarn storybook:dev
+
+# In a browser, navigate to http://localhost:6006/
```
Comment (cpcramer) [inline site/docs/pages/guides/contribution.mdx:77]: We can leave this out since `yarn storybook:dev` auto directs the user to that url

File: site/docs/pages/guides/contribution.mdx [hunk 2]
```diff
@@ -65,6 +65,18 @@ onchainkit
 
 ## Workflows
 
+### Storybook
```
Comment (cpcramer) [inline site/docs/pages/guides/contribution.mdx:68]: I think that this section needs more step by step guidance on how to contribute to onchainkit's Storybook. I'm new to Storybook, and this section in its current state does not help me understand how contributing to onchainKit / Storybook works.

**Review by Zizzamia — APPROVED** [review]

**Comment by Zizzamia** [comment]
@siddharthkul, I approved and merge for now, but @cpcramer is right, we need a bit more love into this section.
